File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxProduct = function (nums) {
    let max = -1;
    let sMax = -1;

    for (let i = 0; i < nums.length; i++) {
        if (max < nums[i]) {
            sMax = max;
            max = nums[i];
        } else if (sMax < nums[i]) {
            sMax = nums[i];
        }
    }

    let ans = (max - 1) * (sMax - 1);

    return ans;
};

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.java
class Solution {
    public int maxProduct(int[] nums) {
        int fm = -1;
        int sm = -1;

        int n = 0;

        do {
            int val = nums[n];

            if(fm < val) {
                sm = fm;
                fm = val;
            }else {
                if(sm < val) {
                    sm = val;
                }
            }

            n++;
        }while (n < nums.length);

        int ans = (fm - 1) * (sm - 1);

        return ans;
    }
}